feat: hobby deployment goes woke - TUI is here

rafaeelaudibert opened the pull request:
I was just messing around, but it looked cool so I kept going as a side project inside the hackathon and got it to look very very cool! I don't wanna replace the docs/old code yet because I wanna release it in beta in the website first, and only eventually remove the old snippet.  
  
It supports setup/update out of the box and has some quirky easter eggs.

[Screen Recording 2025-12-11 at 00.11.37.mov <span class="graphite__hidden">(uploaded via Graphite)</span> <img class="graphite__hidden" src="https://app.graphite.com/user-attachments/thumbnails/2be71280-5204-4755-a065-060013b0b7b6.mov" />](https://app.graphite.com/user-attachments/video/2be71280-5204-4755-a065-060013b0b7b6.mov)



If you wanna try this out you can create a new Ubuntu 24 box on AWS and run these:

```
curl -fsSL https://raw.githubusercontent.com/posthog/posthog/rafa/hobby-deployment-goes-woke/bin/posthog-hobby -o posthog-hobby
chmod +x posthog-hobby
./posthog-hobby
```


I gotta rename this eventually but I'll just keep it as something cryptic for now

rafaeelaudibert committed:
feat: hobby deployment goes woke - TUI is here

I was just messing around, but it looks cool, still have to test it for real in an Ubuntu box

rafaeelaudibert committed:
chore: build posthog-hobby

rafaeelaudibert committed:
feat: Add logger

This will make the UI much cleaner and will let us keep track of what's going on much more easily

rafaeelaudibert committed:
feat: Simplify install/update to a single flow

We now autodetect whether we're on install/update mode and automatically start the adequate process

Also add a hedgehog to the homescreen

rafaeelaudibert committed:
refactor: Move hedgehog to separate class

rafaeelaudibert committed:
fix: Make it more obvious you should press enter

rafaeelaudibert committed:
refactor: Improve UI from welcome screen

Give more space to stuff

rafaeelaudibert committed:
feat: Copy more features from `hobby-install` script

rafaeelaudibert committed:
feat: Use `posthog-go`

rafaeelaudibert committed:
feat: Create new CI mode

Rather than having a full-blown TUI this is a simpler flow that only logs messages

rafaeelaudibert committed:
docs: Improve README

rafaeelaudibert committed:
build: Add new built image

rafaeelaudibert committed:
fix: Add new scripts to `manifest.yaml`

rafaeelaudibert committed:
refactor: use builtin function

rafaeelaudibert committed:
fix: Check os.Chmod errors

rafaeelaudibert committed:
security: Move from math/rand to crypto/rand

rafaeelaudibert committed:
security: Move from 224 to 384 bits secret

rafaeelaudibert committed:
build: build new version

pauldambra reviewed (approved):
oh, i don't get a plugin server even with the bog-standard install

so I'd say 🚢 🚢 🚢 🚢 🚢 

(given installing docker should be a nicer experience)

rafaeelaudibert commented:
@pauldambra I considered installing docker, but... should we? This way, we aren't forcing people to install Docker + Docker compose, we just need something *compatible*. This means, for example, that it works on MacOS where we don't have docker desktop but instead use Orbstack.

We don't install Docker on `bin/deploy-hobby` either

pauldambra commented:
we definitely install docker in the existing script (i've never had to manually install it)
and it requires sudo
so it's in some way nice to not have it
but then we should have a nice exit mode to let people know
(since it's no longer a one shot easy install)

rafaeelaudibert committed:
fix: Use adaptative color rather than hardcoding to white

This broke on light mode

rafaeelaudibert committed:
feat: Add 100 pets easter egg

rafaeelaudibert committed:
chore: Add hobby-installer to .gitignore

avoid commiting local build in case someone uses `go build .` over `make`

rafaeelaudibert committed:
refactor: Make hedgehog more hedgehog-like

It was just a bunny before

rafaeelaudibert committed:
feat: Install/start Docker properly

pauldambra commented:
Now i get

```
root@ubuntu-8gb-nbg1-1-ph:~# curl -fsSL https://raw.githubusercontent.com/posthog/posthog/rafa/hobby-deployment-goes-woke/bin/posthog-hobby -o posthog-hobby
chmod +x posthog-hobby
./posthog-hobby
-bash: ./posthog-hobby: cannot execute binary file: Exec format error
```

rafaeelaudibert commented:
Huh, I'll get back at this next Monday then

rafaeelaudibert commented:
I dont know how that could've happened - the Mac one has a different name locally, but I might have messed something up

rafaeelaudibert committed:
chore: Bump to go v1.24 and add golangci-lint

We are running go v1.24 for the hobby and livestream, make sure the env reflects that

rafaeelaudibert committed:
fix: Cleanup makefile with proper keys

rafaeelaudibert committed:
fix: Add new entry to manifest

rafaeelaudibert committed:
fix: Run lint and add action for it

rafaeelaudibert committed:
chore: Build `posthog-hobby`

rafaeelaudibert committed:
fix: Fix golangci action to specific hash

rafaeelaudibert committed:
feat: Add action to rebuild hobby installer in case of changes

This will make sure we don't fail to do that and have outdated installation

rafaeelaudibert committed:
refactor: Remove useless/incorrect funFacts

rafaeelaudibert committed:
fix: Fix manifest

rafaeelaudibert committed:
fix: Wait for docker for longer

rafaeelaudibert committed:
fix: Check sudo is required when installing docker

rafaeelaudibert committed:
refactor: Add more useful message

rafaeelaudibert committed:
feat: Hide telemetry steps from view

rafaeelaudibert committed:
refactor: Move from checking sudo to checking root

rafaeelaudibert committed:
refactor: Wait up to 30min

rafaeelaudibert committed:
refactor: Improved messages when installing docker

rafaeelaudibert committed:
refactor: Stop mentioning option

It's automatic

rafaeelaudibert committed:
feat: Add debug log file

rafaeelaudibert commented:
@pauldambra let's fucking go

![image.png](https://app.graphite.com/user-attachments/assets/17c3ec7e-2b8a-479a-bd14-171fc28d8923.png)

rafaeelaudibert committed:
fix: Fix linter warnings

pauldambra commented:
ugh... and we should shift the hobby smoke test job that deploys to digital ocean to this script
(although not blocking, just tidier)

rafaeelaudibert commented:
@pauldambra Yeah, was just double-tripe-quadruple checking before moving the hobby deploy script to this.

I didn't try updating something installed via the original script but it *should* just work, slightly unusual that it was downloading from Docker on the "Start PostHog stack" step, it definitely doesn't look like that when I'm running it myself on an empty box/follow-up upgrade

rafaeelaudibert committed:
feat: Point `hobby-ci` deployment script to the new `posthog-hobby` binary
